### PR TITLE
Review Suggestions from #13

### PR DESCRIPTION
**Pull request by rishiip**
Review modifications that are suggested on merged PR #13 

SCR-20220825-mr3
SCR-20220825-mrg

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/constants.jsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import React from "react";
 import JacobDisplayPicture from "images/JacobDisplayPicture";
 import { MenuHorizontal } from "neetoicons";
 import { Avatar, Typography } from "neetoui";
+import * as Yup from "yup";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/constants.jsx:6]: yup should be lowercase. [Refer](https://academy.bigbinary.com/learn-react/pages/miscellaneous-tips#:~:text=Before%20we%20move%20ahead%20it%20is%20worth%20pointing%20out%20that%20we%20should%20always%20import%20yup%20in%20lowercase%20as%20it%20is%20imported%20as%20an%20object.%20Only%20classes%20and%20components%20should%20be%20imported%20in%20Pascal%20case.%20And%20constants%20should%20be%20imported%20as%20they%20are%20declared.)

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
 };
 
-export const NOTES_FORM_VALIDATION_SCHEMA = yup.object().shape({
-  title: yup.string().required("Title is required"),
-  description: yup.string().required("Description is required"),
+export const VALIDATION_SCHEMA = Yup.object().shape({
+  title: Yup.string().required("Title is required"),
+  description: Yup.string().required("Description is required"),
+  role: Yup.object().required("Role is required"),
+  tags: Yup.array().required("Tag is required"),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:16]: It's always better to also add validation for the inner keys of the objects and arrays.

Refer https://github.com/jquense/yup#getting-started

**Review by yedhink — CHANGES_REQUESTED** [review]
@rishiip Few comments.

File: app/javascript/src/components/Dashboard/Contacts/constants.jsx (hunk 1 above)
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Contacts/constants.jsx]: This is done. As we're using `Yup` in neetoCRM, previously I thought to use the same here with first letter as capital.

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 2 above)
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/constants.js]: This is done.

**Comment by rishiip** [comment]
> @rishiip Few comments.

@yedhink - This is done. Ref [commit](https://github.com/rishiip/wheel/pull/15/commits/78f86883628ba85b36b566c6db0438abfbdd7112).

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 3]
```diff
@@ -2,33 +2,60 @@ import dayjs from "dayjs";
...
+      yup
+        .object()
+        .nullable()
+        .shape({
+          label: yup.string().oneOf(TAGS.map(tag => tag.label)),
+          value: yup.string().oneOf(TAGS.map(tag => tag.value)),
+        })
+    )
+    .nullable()
+    .required("Tag is required"),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:56]: Had you tested this validation from the UI? If no tags are provided does it show an error right now and prevent you from submitting the form as expected?

From my perspective this won't work. But please verify and get back to me.

**Review by yedhink — CHANGES_REQUESTED** [review]
@rishiip _a

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 3 above)
Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/constants.js:56]: @yedhink - 
1. If tags are not provided it's preventing user to submit form as expected. [Have a look.](https://www.loom.com/share/9af7f44f22354455a306729bb410a1d6)
2. Providing `nullable` and `required` to the same entity explanation - [Have a look](https://www.loom.com/share/3a667865efa94141a28c354921411c9b)

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/constants.js:56]: @rishiip https://www.loom.com/share/5d3538062fdd4a42aeaea5275f97c241

Comment (rishiip) [reply app/javascript/src/components/Dashboard/Notes/constants.js]: @yedhink - Thanks for pointing this out. This is done. [Have a look at explanation](https://www.loom.com/share/0ec5d459073f490484c7978e48e9466d).

**Pull request merged**